test/matrix-test.jl

```diff
@@ -59,3 +59,34 @@ end
   @test is_zero(reduce_mod(A, B))
   @test is_zero(reduce_mod(A, 2*B))
 end
+
+@testset "Eigenvalues of ZZMatrix" begin
+  A4 = matrix(ZZ,4,4, [6,-8,9,5,  5,-6,5,0,  0,6,5,4,  -6,8,-1,7]);
+  E4 = eigenvalues(A4);
+  @test length(E4) == 2;
+  @test E4 == [2,4] || E4 == [4,2]
+  E4m = eigenvalues_with_multiplicities(A4);
+  @test length(E4m) == 2;
+  @test E4m[1][2] == 1 || E4m[2][2] == 1
+
+
+  A5 = matrix(ZZ,5,5, [0,0,1,-1,-1,  -1,-1,1,1,-1,  0,0,1,-1,-1,  -1,1,1,-1,-1,  1,1,-1,-1,1])
+  E5 = eigenvalues(A5);
+  @test length(E5) == 3;
+  @test sum(E5) == -1 && sum((e -> e^2), E5) == 5
```

Comment:
```suggestion
  @test sum(E5) == -1
  @test sum((e -> e^2), E5) == 5
```
splitting this into two tests makes it a bit easier to debug